Source/WarpX.cpp

```diff
@@ -554,6 +554,11 @@ WarpX::ReadParameters ()
 
         // Only needs to be set with WARPX_DIM_RZ, otherwise defaults to 1
         pp.query("n_rz_azimuthal_modes", n_rz_azimuthal_modes);
+
+#if defined WARPX_DIM_RZ
+        AMREX_ALWAYS_ASSERT_WITH_MESSAGE(Geom(0).isPeriodic(0) == 0,
```

Comment:
Minor question, do we need to indicate that `Geom()` is from `GetInstance()`?
```suggestion
        AMREX_ALWAYS_ASSERT_WITH_MESSAGE(GetInstance().Geom(0).isPeriodic(0) == 0,
```

Reply:
Good question. Geom is used earlier in this routine and I just copied that code. 'ReadParameters' is not a static routine so it has direct access to the attributes of the WarpX object. Compare this to `CellSize` which is declared static.

Reply:
>  'ReadParameters' is not a static routine so it has direct access to the attributes of the WarpX object.
Thanks for explaining @dpgrote ! That does makes sense.